### ISSUE TITLE
HaxMin not renaming 'com.package.className' strings

### ISSUE THREAD

**Issue by outerate** [YellowAfterlife/HaxMin#8]
A significant number of these strings are not getting renamed.

My minified output contains a bunch of occurrences that look similar to this:
 
```js
var pR = function () {};
qO["jF.myfolder.MyClassName"] = pR;
pR.b4 = "jF.myfolder.MyClassName";
pR.bq = true;
pR.prototype = {
    d2: pR
};
```

I tried troubleshooting this by looking in "default.txt", my "whitelist.txt", "Main.hx", and "HaxeMin.hx", for the strings "myfolder" and "MyClassName", but I find no matches anywhere.

"myfolder" and "MyClassName" are just example strings. I used the actually strings from my code for the above step.

TL;DR - HaxMin is behaving as if many of my `com.package.className` strings have been whitelisted, but they definitely aren't, and it's preventing the desired minification from occurring.

**Comment by outerate** [comment]
Here is a simple example that reproduces the issue:

```haxe
// src/myexamplefolder/MyExampleClass.hx

package myexamplefolder;

import haxe.ds.ArraySort;

class MyExampleClass {
    public static function main() {
        final foo = [1, 2, 20, 3, -1];
        ArraySort.sort(foo, (a, b) -> {
            if(a < b) return -1;
            if(a > b) return 1;
            else return 0;
        });
        trace(foo);

        final AS = Type.resolveClass('haxe.ds.ArraySort');
        trace(AS);
    }
}
```

```sh
$ haxe --version
4.2.3
$ haxe -cp src -main myexamplefolder.MyExampleClass -js ./test.js
$ neko Haxmin.n test.js test.min.js
```

## JavaScript before minification:
```js
(function ($global) { "use strict";
var $hxClasses = {};
var haxe_ds_ArraySort = function() { };
$hxClasses["haxe.ds.ArraySort"] = haxe_ds_ArraySort;
haxe_ds_ArraySort.sort = function(a,cmp) {
	haxe_ds_ArraySort.rec(a,cmp,0,a.length);
};
haxe_ds_ArraySort.rec = function(a,cmp,from,to) {
	var middle = from + to >> 1;
	if(to - from < 12) {
		if(to <= from) {
			return;
		}
		var _g = from + 1;
		var _g1 = to;
		while(_g < _g1) {
			var i = _g++;
			var j = i;
			while(j > from) {
				if(cmp(a[j],a[j - 1]) < 0) {
					haxe_ds_ArraySort.swap(a,j - 1,j);
				} else {
					break;
				}
				--j;
			}
		}
		return;
	}
	haxe_ds_ArraySort.rec(a,cmp,from,middle);
	haxe_ds_ArraySort.rec(a,cmp,middle,to);
	haxe_ds_ArraySort.doMerge(a,cmp,from,middle,to,middle - from,to - middle);
};
haxe_ds_ArraySort.doMerge = function(a,cmp,from,pivot,to,len1,len2) {
	var first_cut;
	var second_cut;
	var len11;
	var len22;
	if(len1 == 0 || len2 == 0) {
		return;
	}
	if(len1 + len2 == 2) {
		if(cmp(a[pivot],a[from]) < 0) {
			haxe_ds_ArraySort.swap(a,pivot,from);
		}
		return;
	}
	if(len1 > len2) {
		len11 = len1 >> 1;
		first_cut = from + len11;
		second_cut = haxe_ds_ArraySort.lower(a,cmp,pivot,to,first_cut);
		len22 = second_cut - pivot;
	} else {
		len22 = len2 >> 1;
		second_cut = pivot + len22;
		first_cut = haxe_ds_ArraySort.upper(a,cmp,from,pivot,second_cut);
		len11 = first_cut - from;
	}
	haxe_ds_ArraySort.rotate(a,cmp,first_cut,pivot,second_cut);
	var new_mid = first_cut + len22;
	haxe_ds_ArraySort.doMerge(a,cmp,from,first_cut,new_mid,len11,len22);
	haxe_ds_ArraySort.doMerge(a,cmp,new_mid,second_cut,to,len1 - len11,len2 - len22);
};
haxe_ds_ArraySort.rotate = function(a,cmp,from,mid,to) {
	if(from == mid || mid == to) {
		return;
	}
	var n = haxe_ds_ArraySort.gcd(to - from,mid - from);
	while(n-- != 0) {
		var val = a[from + n];
		var shift = mid - from;
		var p1 = from + n;
		var p2 = from + n + shift;
		while(p2 != from + n) {
			a[p1] = a[p2];
			p1 = p2;
			if(to - p2 > shift) {
				p2 += shift;
			} else {
				p2 = from + (shift - (to - p2));
			}
		}
		a[p1] = val;
	}
};
haxe_ds_ArraySort.gcd = function(m,n) {
	while(n != 0) {
		var t = m % n;
		m = n;
		n = t;
	}
	return m;
};
haxe_ds_ArraySort.upper = function(a,cmp,from,to,val) {
	var len = to - from;
	var half;
	var mid;
	while(len > 0) {
		half = len >> 1;
		mid = from + half;
		if(cmp(a[val],a[mid]) < 0) {
			len = half;
		} else {
			from = mid + 1;
			len = len - half - 1;
		}
	}
	return from;
};
haxe_ds_ArraySort.lower = function(a,cmp,from,to,val) {
	var len = to - from;
	var half;
	var mid;
	while(len > 0) {
		half = len >> 1;
		mid = from + half;
		if(cmp(a[mid],a[val]) < 0) {
			from = mid + 1;
			len = len - half - 1;
		} else {
			len = half;
		}
	}
	return from;
};
haxe_ds_ArraySort.swap = function(a,i,j) {
	var tmp = a[i];
	a[i] = a[j];
	a[j] = tmp;
};
var haxe_iterators_ArrayIterator = function(array) {
	this.current = 0;
	this.array = array;
};
$hxClasses["haxe.iterators.ArrayIterator"] = haxe_iterators_ArrayIterator;
haxe_iterators_ArrayIterator.prototype = {
	hasNext: function() {
		return this.current < this.array.length;
	}
	,next: function() {
		return this.array[this.current++];
	}
};
var myexamplefolder_MyExampleClass = function() { };
$hxClasses["myexamplefolder.MyExampleClass"] = myexamplefolder_MyExampleClass;
myexamplefolder_MyExampleClass.main = function() {
	var foo = [1,2,20,3,-1];
	haxe_ds_ArraySort.sort(foo,function(a,b) {
		if(a < b) {
			return -1;
		}
		if(a > b) {
			return 1;
		} else {
			return 0;
		}
	});
	console.log("src/myexamplefolder/MyExampleClass.hx:15:",foo);
	var AS = $hxClasses["haxe.ds.ArraySort"];
	console.log("src/myexamplefolder/MyExampleClass.hx:18:",AS);
};
$hxClasses["Math"] = Math;
$hxClasses["Array"] = Array;
myexamplefolder_MyExampleClass.main();
})({});
```

## JavaScript after minification:
```js
 (function (h) {
     "use strict";
     var q = {};
     var U = function () {};
     q["haxe.ds.ArraySort"] = U;
     U.sort = function (a, cmp) {
         U.o(a, cmp, 0, a.length)
     };
     U.o = function (a, cmp, P, to) {
         var w = P + to >> 1;
         if (to - P < 12) {
             if (to <= P) {
                 return
             };
             var V = P + 1;
             var Y = to;
             while (V < Y) {
                 var I = V++;
                 var m = I;
                 while (m > P) {
                     if (cmp(a[m], a[m - 1]) < 0) {
                         U.l(a, m - 1, m)
                     } else {
                         break
                     }--m
                 }
             };
             return
         };
         U.o(a, cmp, P, w);
         U.o(a, cmp, w, to);
         U.M(a, cmp, P, w, to, w - P, to - w)
     };
     U.M = function (a, cmp, P, $, to, j, F) {
         var R;
         var D;
         var W;
         var X;
         if (j == 0 || F == 0) {
             return
         };
         if (j + F == 2) {
             if (cmp(a[$], a[P]) < 0) {
                 U.l(a, $, P)
             };
             return
         };
         if (j > F) {
             W = j >> 1;
             R = P + W;
             D = U.lower(a, cmp, $, to, R);
             X = D - $
         } else {
             X = F >> 1;
             D = $ + X;
             R = U.upper(a, cmp, P, $, D);
             W = R - P
         };
         U.rotate(a, cmp, R, $, D);
         var g = R + X;
         U.M(a, cmp, P, R, g, W, X);
         U.M(a, cmp, g, D, to, j - W, F - X)
     };
     U.rotate = function (a, cmp, P, N, to) {
         if (P == N || N == to) {
             return
         };
         var v = U.t(to - P, N - P);
         while (v-- != 0) {
             var k = a[P + v];
             var shift = N - P;
             var n = P + v;
             var T = P + v + shift;
             while (T != P + v) {
                 a[n] = a[T];
                 n = T;
                 if (to - T > shift) {
                     T += shift
                 } else {
                     T = P + (shift - (to - T))
                 }
             };
             a[n] = k
         }
     };
     U.t = function (G, v) {
         while (v != 0) {
             var J = G % v;
             G = v;
             v = J
         };
         return G
     };
     U.upper = function (a, cmp, P, to, k) {
         var S = to - P;
         var K;
         var N;
         while (S > 0) {
             K = S >> 1;
             N = P + K;
             if (cmp(a[k], a[N]) < 0) {
                 S = K
             } else {
                 P = N + 1;
                 S = S - K - 1
             }
         };
         return P
     };
     U.lower = function (a, cmp, P, to, k) {
         var S = to - P;
         var K;
         var N;
         while (S > 0) {
             K = S >> 1;
             N = P + K;
             if (cmp(a[N], a[k]) < 0) {
                 P = N + 1;
                 S = S - K - 1
             } else {
                 S = K
             }
         };
         return P
     };
     U.l = function (a, I, m) {
         var p = a[I];
         a[I] = a[m];
         a[m] = p
     };
     var C = function (H) {
         this.O = 0;
         this.H = H
     };
     q["haxe.iterators.ArrayIterator"] = C;
     C.prototype = {
         _: function () {
             return this.O < this.H.length
         },
         i: function () {
             return this.H[this.O++]
         }
     };
     var u = function () {};
     q["myexamplefolder.MyExampleClass"] = u;
     u.B = function () {
         var L = [1, 2, 20, 3, -1];
         U.sort(L, function (a, b) {
             if (a < b) {
                 return -1
             };
             if (a > b) {
                 return 1
             } else {
                 return 0
             }
         });
         console.log("src/myexamplefolder/MyExampleClass.hx:15:", L);
         var A = q["haxe.ds.ArraySort"];
         console.log("src/myexamplefolder/MyExampleClass.hx:18:", A)
     };
     q["Math"] = Math;
     q["Array"] = Array;
     u.B()
 })({});
```

## Strings expected to get renamed that weren't:

```haxe
q["haxe.ds.ArraySort"] = U;
q["haxe.iterators.ArrayIterator"] = C;
q["myexamplefolder.MyExampleClass"] = u;
```

Inspecting HaxMin.hx leads me to believe this should be working?

https://github.com/YAL-Haxe/HaxMin/blob/ca0cd6e9bc44237b7740c562947ab114aef8e73e/src/Haxmin.hx#L394-L397
